unixfs/io/dagreader.go

```diff
@@ -19,14 +20,29 @@ var ErrIsDir = errors.New("this dag node is a directory")
 type DagReader struct {
 	serv         mdag.DAGService
 	node         *mdag.Node
-	buf          io.Reader
-	fetchChan    <-chan *mdag.Node
+	pbdata       *ftpb.Data
+	buf          ReadSeekCloser
+	promises     []mdag.NodeGetter
 	linkPosition int
+	offset       int64
+
+	// Our context
+	ctx context.Context
+
+	// Context for children
+	fctx   context.Context
+	cancel func()
```

Comment:
Yeah, i would use goprocess, but the contexts main purpose is actually to derive children from for the sub-readers